packages/driver/cypress/e2e/commands/cookies.cy.js

```diff
@@ -9,148 +9,142 @@ describe('src/cy/commands/cookies - no stub', () => {
-        cy.clearCookies({ domain: null })
-      }
+      cy.visit('http://www.barbaz.com:3500/fixtures/generic.html')
+
+      cy.setCookie('key1', 'value1', { domain: 'www.foobar.com' })
+      cy.setCookie('key2', 'value2', { domain: 'foobar.com' })
+      cy.setCookie('key3', 'value3') // defaults to hostname: www.barbaz.com
+      cy.setCookie('key4', 'value4', { domain: 'www.barbaz.com' })
+      cy.setCookie('key5', 'value5', { domain: '.www.barbaz.com' })
+      cy.setCookie('key6', 'value6', { domain: 'barbaz.com' })
     })
 
     it('returns cookies from the domain matching the AUT by default', () => {
-      cy.visit('http://www.barbaz.com:3500/fixtures/generic.html')
-      cy.setCookie('foo', 'bar', { domain: 'www.foobar.com' })
-      cy.setCookie('baz', 'qux', { domain: 'foobar.com' })
-      cy.setCookie('foo', 'bar') // defaults to (super)domain: barbaz.com
-      cy.setCookie('qux', 'quuz', { domain: 'www.barbaz.com' })
-
       cy.getCookies().then((cookies) => {
-        expect(cookies).to.have.length(2)
-        // both the barbaz.com and www.barbaz.com cookies are yielded
-        expect(cookies[0].domain).to.match(/\.?barbaz\.com/)
-        expect(cookies[1].domain).to.match(/\.?www\.barbaz\.com/)
+        expect(cookies).to.have.length(3)
+
+        // order is not consistent between browsers or even guaranteed between runs
+        const sortedCookies = Cypress._.sortBy(cookies, 'name')
+
+        expect(sortedCookies[0].name).to.equal('key3')
+        expect(sortedCookies[0].domain).to.match(/\.?www\.barbaz\.com/)
+        expect(sortedCookies[1].name).to.equal('key4')
+        expect(sortedCookies[1].domain).to.match(/\.?www\.barbaz\.com/)
+        expect(sortedCookies[2].name).to.equal('key5')
+        expect(sortedCookies[2].domain).to.match(/\.?www\.barbaz\.com/)
       })
 
-      if (isWebkit || !Cypress.config('experimentalSessionAndOrigin')) return
+      // webkit does not support cy.origin()
+      if (isWebkit) return
 
       cy.origin('http://www.foobar.com:3500', () => {
         cy.visit('http://www.foobar.com:3500/fixtures/generic.html')
 
         cy.getCookies().then((cookies) => {
+          expect(cookies).to.have.length(1)
+          expect(cookies[0].name).to.equal('key1')
           expect(cookies[0].domain).to.match(/\.?www\.foobar\.com/)
-          expect(cookies[1].domain).to.match(/\.?foobar\.com/)
         })
       })
     })
 
     it('returns cookies for the specified domain', () => {
-      cy.visit('http://www.barbaz.com:3500/fixtures/generic.html')
-      cy.setCookie('foo', 'bar', { domain: 'www.foobar.com' })
-      cy.setCookie('foo', 'bar') // defaults to (super)domain: barbaz.com
-      cy.setCookie('qux', 'quuz', { domain: 'www.barbaz.com' })
-
       cy.getCookies({ domain: 'www.foobar.com' }).then((cookies) => {
+        expect(cookies).to.have.length(1)
+        expect(cookies[0].name).to.equal('key1')
         expect(cookies[0].domain).to.match(/\.?www\.foobar\.com/)
```

Comment:
Even in the case where the domain is specified explicitly, I would expect all applicable cookies in that domain to be returned, which would include the higher level domain
```suggestion
      cy.getCookies({ domain: 'www.foobar.com' }).then((cookies) => {
        expect(cookies).to.have.length(2)
        expect(cookies[0].name).to.equal('key1')
        expect(cookies[0].domain).to.match(/\.?www\.foobar\.com/)
        expect(cookies[0].name).to.equal('key2')
        expect(cookies[0].domain).to.match(/\.?foobar\.com/)
```